src/pip/_internal/resolution/base.py

```diff
@@ -0,0 +1,20 @@
+from pip._internal.utils.typing import MYPY_CHECK_RUNNING
+
+if MYPY_CHECK_RUNNING:
+    from typing import Callable, List
+    from pip._internal.req.req_install import InstallRequirement
+    from pip._internal.req.req_set import RequirementSet
+
+    InstallRequirementProvider = Callable[
+        [str, InstallRequirement], InstallRequirement
+    ]
+
+
+class BaseResolver(object):
```

Comment:
Honestly I dislike `abc.ABCMeta` in general since it adds unnecessary runtime overhead. But I’d be happy to fix this if there is a strong project preference to use it.

Reply:
I'm happy to leave this as it is. I don't particularly see much value in `ABCMeta`.

Reply:
It was nit anyway, either works for me too!